### ISSUE TITLE
Make url changes more smooth

### ISSUE THREAD

**Issue by victor-rocha** [codefordurham/school-navigator#64]
performing a search triggers a URL change and one can notice the map flickering while it updates. 

http://stackoverflow.com/questions/24264303/template-flicker-when-url-changes-in-angular

**Comment by copelco** [comment]
@nickpassarella can you push up your branch for @victor-rocha to review?

**Comment by nickpassarella** [comment]
I know I have been looking at this forever, but I think there is an issue inherent to Angular with causing a page reload when the url changes. I've hacked together several pieces that have almost worked. I think the right path will be to format the url with latitude and longitude as a query in the url like location?lat=0000&lng=0000 instead of location/lat/long. With that, we can set reloadOnSearch=false, and it should change the url (which will still be shareable) without causing a page reload. I will dig into that option over the next couple of days.

**Comment by lemanal** [comment]
Have you looked at this again @nickpassarella ?

**Comment by NoOutlet** [comment]
I am very close to having this completed.
The way I did it was to refactor basically the entire Angular app using [angular-leaflet-directive](https://github.com/tombatossals/angular-leaflet-directive). To make it easy for me to follow the codeflow, I grouped everything (except templates) into a single file app/app.js. Hence the name of the branch is "oneFile".

I believe the only two things I have not worked out yet are the autosetting of the height of the map widget and the autosetting of the height of the school list. The "legacy" code for those functions are in app/app.js.
Merging this in will solve this issue (#64) as well as issue #130. It also drastically improves performance of certain features (notably showing districts on hover of school marker or list item).

I may have to hand this issue over to someone as I am expecting a second child in the next two weeks. Also, this refactor is nearly a complete rewrite of the app and should be pretty well-tested before merging it in, I think.
